hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/fs/viewfs/TestViewFileSystemHdfs.java

```diff
@@ -506,4 +508,29 @@ public void testInternalDirectoryPermissions() throws IOException {
     assertEquals(fs.getFileStatus(subDirOfInternalDir).getPermission(),
         fs.getFileStatus(subDirOfRealDir).getPermission());
   }
+
+  @Test
+  public void testEnclosingRootsBase() throws Exception {
+    final Path zone = new Path("/data/EZ");
+    fsTarget.mkdirs(zone);
+    final Path zone1 = new Path("/data/EZ/zone1");
+    fsTarget.mkdirs(zone1);
+
+    DFSTestUtil.createKey("test_key", cluster, 0, CONF);
+    HdfsAdmin hdfsAdmin = new HdfsAdmin(cluster.getURI(0), CONF);
+    final EnumSet<CreateEncryptionZoneFlag> provisionTrash =
+        EnumSet.of(CreateEncryptionZoneFlag.PROVISION_TRASH);
+    hdfsAdmin.createEncryptionZone(zone1, "test_key", provisionTrash);
+    RemoteIterator<EncryptionZone> zones = hdfsAdmin.listEncryptionZones();
+    assertEquals(fsView.getEnclosingRoot(zone), new Path("/data"));
+    assertEquals(fsView.getEnclosingRoot(zone1), zone1);
+
+    Path nn02Ez = new Path("/mountOnNn2/EZ");
+    fsTarget2.mkdirs(nn02Ez);
+    assertEquals(fsView.getEnclosingRoot((nn02Ez)), new Path("/mountOnNn2"));
+    HdfsAdmin hdfsAdmin2 = new HdfsAdmin(cluster.getURI(1), CONF);
+    DFSTestUtil.createKey("test_key", cluster, 1, CONF);
+    hdfsAdmin2.createEncryptionZone(nn02Ez, "test_key", provisionTrash);
+    assertEquals(fsView.getEnclosingRoot((nn02Ez)), nn02Ez);
```

Comment:
nit, maybe test a path which is several levels deeper than the zone?

Reply:
I don't think an encryption zone can span multiple mount points

Reply:
what I wanted to suggest is to test a file which is stored inside the zone, vs, just test the zone dir itself.

So, instead of 

`assertEquals(fsView.getEnclosingRoot((nn02Ez)), nn02Ez);
`
could we do

`assertEquals(fsView.getEnclosingRoot((new Path(nn02Ez, "dir/dir2/file")), nn02Ez);`

Reply:
done